src/platform/test.rs

```diff
@@ -413,6 +413,31 @@ fn receiver_set() {
     }
 }
 
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn sigterm_handler(_: libc::c_int) -> () {}
+
+#[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (tx0, rx0) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx0).unwrap();
+    let parent_pid = unsafe {
+        libc::getpid()
+    };
+    unsafe {
+        libc::signal(libc::SIGTERM,
+                     sigterm_handler as fn(libc::c_int) as *mut libc::c_void as libc::sighandler_t);
+    }
+    let _ = unsafe { fork(|| {
```

Comment:
>  I wonder whether you could send the signal the other way instead

Overall, I like sending the signal to the child better. It could make debugging a little harder if an unexpected panic occurs in the `select`. Overall much better though.